pkg/kubelet/kubelet_pods.go

```diff
@@ -86,24 +86,115 @@ func (kl *Kubelet) GetActivePods() []*v1.Pod {
 	return activePods
 }
 
+func (kl *Kubelet) parseDeviceSpec(deviceSpec string) (deviceInfo kubecontainer.DeviceInfo, err error) {
+	if len(deviceSpec) == 0 {
+		err = fmt.Errorf("device spec is empty")
+		return
+	}
+
+	deviceArr := strings.SplitN(deviceSpec, ":", 3)
+	if !filepath.IsAbs(deviceArr[0]) {
+		err = fmt.Errorf("incorrect value of PathOnHost in device spec: %s", deviceSpec)
+		return
+	}
+
+	// PathOnHost
+	deviceInfo.PathOnHost = deviceArr[0]
+
+	// PathInContainer
+	if len(deviceArr) >= 2 && filepath.IsAbs(deviceArr[1]) {
+		deviceInfo.PathInContainer = deviceArr[1]
+	} else {
+		deviceInfo.PathInContainer = deviceInfo.PathOnHost
+	}
+
+	// Permissions
+	if len(deviceArr) == 3 && len(deviceArr[2]) <= 3 && strings.ContainsAny(deviceArr[2], "mrw") {
+		deviceInfo.Permissions = deviceArr[2]
+	} else {
+		deviceInfo.Permissions = "mrw"
+	}
+
+	return
+}
+
+// Device spec:
+//  - PathOnHost:PathInContainer:Permissions
+//  - /dev/kvm:/dev/kvm:mrw     (full spec)
+//  - /dev/kvm:/dev/kvm-new:rw  (/dev/kvm -> /dev/kvm-new without mknod)
+//  - /dev/kvm::mrw             (device will be mapped as one to one)
+//  - /dev/kvm                  (device will be mapped as one to one, and will be applied default permissions: "mrw")
+// Annotations example:
+//  - Full spec:
+//      device.beta.kubernetes.io/allow-list: "/dev/kvm:/dev/kvm:mrw;/dev/fuse:/dev/fuse:mrw"
+//  - Short spec:
+//      device.beta.kubernetes.io/allow-list: "/dev/kvm;/dev/fuse"
+
+/*
+apiVersion: v1
+kind: Pod
+metadata:
+  name: alpine
+  labels:
+    app: alpine
+  annotations:
+    device.beta.kubernetes.io/allow-list: "/dev/kvm;/dev/fuse"
```

Comment:
What happens if I have multiple containers in my pod and I only want one to mount this hostPath?